Style submenus container

JustinyAhin opened the pull request:
<!-- Add reference to the issue this pull-request fixes.-->
Fixes #660 & #680.

## Summary
Style primary submenu container.

## Relevant technical choices:
No

## Test instructions

This PR can be tested by following these 

1. Add a menu to the theme which contain submenus
2. Hover over the menus items with ths "+" symbol

## Screenshots
![Capture d’écran du 2020-10-26 09-59-12](https://user-images.githubusercontent.com/33403964/97152934-f9863e80-1771-11eb-9f8a-2abaf0c7a990.png)
![Capture d’écran du 2020-10-26 09-56-10](https://user-images.githubusercontent.com/33403964/97152936-fab76b80-1771-11eb-9692-becbd5f4fd7f.png)
![Capture d’écran du 2020-10-26 09-55-29](https://user-images.githubusercontent.com/33403964/97152940-fb500200-1771-11eb-86a9-36449ce72264.png)



## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

JustinyAhin committed:
Style primary submenu container

JustinyAhin commented:
Ping @joyously

joyously commented:
Looks better to me! but it doesn't match the [mockup](https://github.com/WordPress/twentytwentyone/issues/660#issuecomment-715451116) I saw, which had the border around the parent item also (I don't know if that matters). 
In the dark version the border seems barely discernible.
Can you show a Navigation block screenshot? (your test instructions are not coherent)
That [mockup](https://github.com/WordPress/twentytwentyone/issues/680#issue-728488907) seems different.

JustinyAhin committed:
Add border to navigation menu block

JustinyAhin committed:
Update submenus border color

JustinyAhin commented:
>  In the dark version the border seems barely discernible.

I've updated the fix, and added it to the navigation block too.

What I did basically to both primary menus and navigation blocks that have submenus is adding a 1px solid light-gray (#39414d) border to them.

It looks fine for both with the default background color, but could be more visible on darker background colors (I'm not sure we can handle this before correctly supporting dark mode in the theme -- cc @aristath ).

Primary menu on default light background
![Capture d’écran du 2020-10-26 15-08-13](https://user-images.githubusercontent.com/33403964/97183209-c8bbfe80-179d-11eb-8e0c-eb74e07f2b9f.png)

Primary menu on dark background
![Capture d’écran du 2020-10-26 15-07-11](https://user-images.githubusercontent.com/33403964/97183230-d1143980-179d-11eb-8c17-121a8c240512.png)

Navigation block on default light background
![Capture d’écran du 2020-10-26 15-08-30](https://user-images.githubusercontent.com/33403964/97183406-0882e600-179e-11eb-8604-d7075119d2d4.png)


Navigation block on dark background
![Capture d’écran du 2020-10-26 15-07-31](https://user-images.githubusercontent.com/33403964/97183349-f608ac80-179d-11eb-8b7f-ba5f7c0a227a.png)

melchoyce commented:
Thanks for working on this, @JustinyAhin. Can you update the design to look like this?

![image](https://user-images.githubusercontent.com/2846578/97183892-acf03280-1774-11eb-8c8b-0e5cfd85b0c6.png)
[Figma Mockup](https://www.figma.com/file/OEtsCj8oD8JPVx4OvsS0U2/Twenty-Twenty-One?node-id=921%3A38)

JustinyAhin commented:
@melchoyce best I think can be done is something like this:
![Capture d’écran du 2020-10-26 15-56-11](https://user-images.githubusercontent.com/33403964/97188762-29e6d080-17a4-11eb-8c27-a72c8a86cb90.png)

Because the submenu `<ul>` is larger than the menu item, it's a bit tricky to make the border-top disappear, while leaving a small part for the overflow.

There is also a slight offset to the right of the submenu, but I can find out where it comes from, and undo it.

aristath commented:
Thinking out loud:
Normally I'd suggest removing the bottom border from the parent, then changing its z-index so it's above the submenu, and finally moving the submenu up by 1px. That _should_ make it appear as if they are one like in the mockup.
But then we may have a problem if the submenu is not as wide as the parent element. Unless we also add min-width:100% to the submenu? :thinking: 
Should be possible to accomplish though I haven't tried it yet so there may be unforeseen complications with all of the above I described

melchoyce commented:
All good points. Would something like this be any less complicated?

![image](https://user-images.githubusercontent.com/2846578/97232132-011a0780-17b3-11eb-991c-47159da855ff.png)

ryelle committed:
Add a border & top arrow to the primary navigation

ryelle committed:
Add the border & top arrow style to navigation blocks, remove dropshadow

ryelle committed:
Add underline on hover to navigation block

ryelle committed:
Build files

ryelle commented:
I've pushed an attempt at the arrow design to this PR. I created a new variable for the border color which inherits from the global color (so it'll flip to white when a dark background is picked). The border only applies to the 2nd level menu, so we don't have nested borders.

![Screen Shot 2020-10-26 at 6 44 11 PM](https://user-images.githubusercontent.com/541093/97236581-c36dac80-17bb-11eb-9eac-0d8b8bca6b5e.png)
![Screen Shot 2020-10-26 at 6 43 15 PM](https://user-images.githubusercontent.com/541093/97236582-c36dac80-17bb-11eb-8ff9-d315f88151da.png)

Navigation block:

It should match the primary nav now. I removed the shadow, and added in the hover style (which was missing). I didn't change anything for the editor style, I think that can be done in another PR if necessary.

![Screen Shot 2020-10-26 at 6 44 32 PM](https://user-images.githubusercontent.com/541093/97236580-c2d51600-17bb-11eb-83f9-22b718161ec9.png)

carolinan commented:
Both the menu and block looks good to me.
